s3/session_test.go

```diff
@@ -0,0 +1,49 @@
+package s3
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/stretchr/testify/assert"
+	"github.com/aws/aws-sdk-go/aws/defaults"
+	"testing"
+)
+
+var bucket = "s3://test-bucket/wal-g-test-folder/Sub0"
+var settings = map[string]string{
+	EndpointSetting: "http://s3.mdst.yandex.net/",
+}
+var config = defaults.Get().Config.WithRegion(settings[RegionSetting])
+
+func TestGetAWSRegionWithEmptyEndpoint(t *testing.T) {
+	findBucketRegion = func(bucket string, config *aws.Config) (string, error) {
```

Comment:
Maybe I missed the point, but I think that we just need to provide different `aws.Config` and compare the result with the expected one? Basically, write unit tests for these functions: 
1. `findBucketRegion` 
2. The new function which extracts the region from the hostname